src/LodeRunner.Core/Models/TestRunPayload.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using LodeRunner.Core.SchemaFilters;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace LodeRunner.Core.Models
+{
+    /// <summary>
+    /// TestRun Payload.
+    /// </summary>
+    [SwaggerSchemaFilter(typeof(TestRunPayloadSchemaFilter))]
+    public class TestRunPayload : BasePayload, INotifyPropertyChanged
```

Comment:
Let's discuss the need for INotifyPropertyChanged.

Reply:
After having refactor and introduced generics to BasePayload, did not go back to review initial approach, we do not need to derive from INotifyPropertyChanged anymore

Reply:
@gortegaMS

Reply:
Refactor BasePayload and updated DataDictionary